freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/AvailableStackProvider.java

```diff
@@ -0,0 +1,45 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.stack.StackService;
+
+@Component
+public class AvailableStackProvider {
```

Comment:
I would do the filtering on db side with a query so we won't move the unnecessary amount of data to the app and build object from them just to drop it right away

Reply:
good suggestion!